sessiontxn/staleread/provider.go

```diff
@@ -149,3 +149,32 @@ func (p *StalenessTxnContextProvider) AdviseWarmup() error {
 func (p *StalenessTxnContextProvider) AdviseOptimizeWithPlan(_ interface{}) error {
 	return nil
 }
+
+// GetReadSnapshot get snapshot with read ts and set the transaction related options
+// before return
+func (p *StalenessTxnContextProvider) GetReadSnapshot() (kv.Snapshot, error) {
+	txn, err := p.sctx.Txn(false)
+	if err != nil {
+		return nil, err
+	}
+
+	if txn.Valid() {
+		return txn.GetSnapshot(), nil
+	}
+
+	sessVars := p.sctx.GetSessionVars()
+	snapshot := sessiontxn.GetSnapshotWithTS(p.sctx, p.ts)
+
+	replicaReadType := sessVars.GetReplicaRead()
+	if replicaReadType.IsFollowerRead() && !sessVars.StmtCtx.RCCheckTS {
```

Comment:
`!sessVars.StmtCtx.RCCheckTS` is not needed

Reply:
right.